src/internal/utils/handle-key-down.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import { KeyCode } from '../keycode';
+import { isRtl } from '../direction';
+
+export default function handleKeyDown({
+  onActivate,
+  onBlockEnd,
+  onBlockStart,
+  onEnd,
+  onHome,
+  onInlineEnd,
+  onInlineStart,
+  onPageDown,
+  onPageUp,
+}: {
+  onActivate?: (event: React.KeyboardEvent) => void;
+  onBlockEnd?: (event: React.KeyboardEvent) => void;
+  onBlockStart?: (event: React.KeyboardEvent) => void;
+  onEnd?: (event: React.KeyboardEvent) => void;
+  onHome?: (event: React.KeyboardEvent) => void;
+  onInlineEnd?: (event: React.KeyboardEvent) => void;
+  onInlineStart?: (event: React.KeyboardEvent) => void;
+  onPageDown?: (event: React.KeyboardEvent) => void;
+  onPageUp?: (event: React.KeyboardEvent) => void;
+}) {
+  return function (event: React.KeyboardEvent) {
+    switch (event.keyCode) {
+      case KeyCode.down:
+        onBlockEnd?.(event);
+        break;
+      case KeyCode.end:
+        onEnd?.(event);
```

Comment:
Are you saying that unlike arrow keys, <kbd>Home</kbd> and <kbd>End</kbd> keys do flip their direction in RTL?

Reply:
My understanding is that the left and right keys should move physically, but the home and end keys (MacOS fn + left/right) should move logically. This means that End should move left in an RTL horizontal overflow even though you are pressing the right key. I am double checking with our resident native Arabic speakers.

Reply:
Make sense for me, but can we have a formal confirmation of this somewhere to be extra sure about our decision?